### PR TITLE
Add latitude, longitude to Address and implement geocoding, fix country default

### PR DESCRIPTION
**Pull request by weedySeaDragon**
PT Story:  Add coordinates to Address class
https://www.pivotaltracker.com/story/show/140865585

(This  is part of what was done in the big, messy, exploratory PR #178.  This replaces part of it; I'll close #178 without merging once I've completely refactored it into smaller PRs and the new PRs are all successfully merged. )

Changes proposed in this pull request:
1. add gecoder gem
2. get the Google API key from the environment (update the env.example file to note it)
1.  add latitude, longitude to migration
2. fix the default for country (was "Sveriges" but should be "Sverige")
3. geocode the address using the most specific info possible
4. add spec tests for geocoding

Ready for review:
@thesuss @patmbolger

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,7 +1,14 @@
...
 # Contact the project admin to get them if you want to deploy some changes to
 # the production server
+
+# for deploying:
 PRODUCTION_SERVER=xxx.xxx.xxx.xxx
 DEPLOY_USER=some_user
 APP_PATH=/path/to/app
 BRANCH=vcs_branch
+
+# used by the system:
```
Comment (patmbolger) [inline .env.example]: Should we note that the real values for these should be placed in `.env` file?

File: app/models/address.rb [hunk 2]
```diff
@@ -16,4 +16,53 @@ class Address < ApplicationRecord
...
+  def geocode_best_possible
+
+    specificity_order = [street_address, post_code, city, sverige_if_nil]
+
+    most_specific = 0
+    least_specific = specificity_order.size - 1
+
+    geo_result = nil
+
+    until most_specific > least_specific || !geo_result.nil?
```
Comment (patmbolger) [inline app/models/address.rb]: Might be a bit easier to read as:
```
until most_specific > least_specific || geo_result.present?
```

**Review by patmbolger — APPROVED** [review]
Nicely done - please see comments.

File: .env.example (hunk 1 above)
Comment (weedySeaDragon) [reply .env.example]: sure.  Good suggestion.  (I'll put a link to the wiki, too.)

File: app/models/address.rb (hunk 2 above)
Comment (weedySeaDragon) [reply app/models/address.rb]: doh! of course.  I just didn't think of .present? for some reason.  Thanks!

**Comment by thesuss** [comment]
```
  def entire_address (is well tested)
    [street_address, city, post_code, sverige_if_nil].compact.join(', ') (seems untested?)
  end
```

**Comment by weedySeaDragon** [comment]
True. :-)  But that's just doing a standard `join` of an array.  I know it changes the testing numbers, but it's really not worth (or appropriate) to test.  (We can count on the standard `Array join` working.)
I put it in a method like that for maintainability and readability:  it makes it really clear what should be changed if the requirements about which items should or shouldn't be used or the order for determining the minimum geolocation. (= readability)
Being is a separate method also means that if a change needs to be made it's nice and separate so that you're unlikely to change something that shouldn't be changed. (= maintainability)

Have I succeeded in making my case? ;-)

**Comment by thesuss** [comment]
`rspec ./spec/models/address_spec.rb:98 # Address gecoding geocode from address` is failing =/

**Pull request merged**